sdks/java/io/azure/src/main/java/org/apache/beam/sdk/io/azure/options/AzureOptions.java

```diff
@@ -0,0 +1,86 @@
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.azure.options;
+
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.util.Configuration;
+import com.azure.identity.DefaultAzureCredentialBuilder;
+import org.apache.beam.sdk.options.Default;
+import org.apache.beam.sdk.options.DefaultValueFactory;
+import org.apache.beam.sdk.options.Description;
+import org.apache.beam.sdk.options.PipelineOptions;
+
+public interface AzureOptions extends PipelineOptions {
+
+  // TODO: Add any other azure options that users should be able to configure
+  // TODO: Confirm that Azure options are in this file, Blobstore options in BlobstoreOptions
+
+  /** The Azure service endpoint used by the Azure client. */
+  @Description("Azure service endpoint used by the Azure client")
+  String getAzureServiceEndpoint();
+
+  void setAzureServiceEndpoint(String value);
+
+  /**
+   * The credential instance that should be used to authenticate against Azure services. The option
+   * value must contain a "@type" field and an Azure credentials provider class as the field value.
+   */
+  @Description(
+      "The credential instance that should be used to authenticate "
+          + "against Azure services. The option value must contain \"@type\" field "
+          + "and an Azure credentials provider class name as the field value.")
+  @Default.InstanceFactory(AzureUserCredentialsFactory.class)
+  TokenCredential getAzureCredentialsProvider();
+
+  void setAzureCredentialsProvider(TokenCredential value);
+
+  /** Attempts to load Azure credentials. */
+  class AzureUserCredentialsFactory implements DefaultValueFactory<TokenCredential> {
+
+    @Override
+    public TokenCredential create(PipelineOptions options) {
+      return new DefaultAzureCredentialBuilder().build();
+    }
+  }
+
+  /** The client configuration instance that should be used to configure Azure service clients. */
+  @Description(
+      "The client configuration instance that should be used to configure Azure service clients")
+  @Default.InstanceFactory(ConfigurationFactory.class)
+  Configuration getClientConfiguration();
+
+  void setClientConfiguration(Configuration configuration);
+
+  /** The client configuration instance that should be used to configure Azure service clients. */
+  @Description(
+      "The client configuration instance that should be used to configure Azure http client configuration parameters."
+          + "Mentioned parameters are the available parameters that can be set. Set only those that need custom changes.")
+  @Default.InstanceFactory(ConfigurationFactory.class)
+  Configuration getAzureHttpConfiguration();
+
+  void setAzureHttpConfiguration(Configuration configuration);
+
+  /** Default Azure client configuration. */
+  class ConfigurationFactory implements DefaultValueFactory<Configuration> {
+
+    @Override
+    public Configuration create(PipelineOptions options) {
+      return new Configuration();
+    }
+  }
```

Comment:
These configuration-related variables seem to be non-serializable. I think we need to implement the special Json serializer before moving forward with adding these changes. We can remove these changes from this PR (and rely purely on a `String`-type connectionString) and add them in a follow-up PR, or get the Json serializer in this PR. What do you think?

Reply:
I was planning to implement a module similar to AwsModule to serialize the non-serializable variables.  I can include it in this PR, I'll work on it today.